pkg/auth/oauth/external/handler.go

```diff
@@ -145,7 +145,7 @@ func DefaultState() State {
 func (defaultState) Generate(w http.ResponseWriter, req *http.Request) (string, error) {
 	state := url.Values{
 		"csrf": {"..."}, // TODO: get csrf
-		"then": {req.URL.String()},
+		"then": {url.QueryEscape(req.URL.String())},
```

Comment:
For maximum compatibility, it might be worth changing our DefaultState impl to base64 encode and decode, just to avoid issues with external providers accidentally decoding state. Still, checking/fixing keycloak state handling as necessary should probably happen as well...